inc/datahandlers/user.php

```diff
@@ -888,6 +888,20 @@ function verify_checkfields()
 	}
 
 	/**
+	 * Verifies if the selected timezone is valid.
+	 *
+	 * @return boolean True when valid, false when invalid.
+	 */
+	function verify_timezone()
+	{
+		$user = &$this->data;
+
+		$timezones = get_supported_timezones();
+
+		return array_key_exists($user['timezone'], $timezones);
```

Comment:
And there are is no option array when the user is created, which is the initial issue here - a default timezone should be set in that case instead of messing with errors... Which shouldn't be printed in ACP as you said yourself in the style PR.